### PR TITLE
🤖  docker table builder

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

Just made a quick tool to make tables with a complete list of the tools and dockers used in our workflows. This was something we discussed doing previously to aid OPs @zhangb1 in tracking differences between releases.

Right now this table builder is only run on production public apps (somatic and consensus).

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Successfully made this PR: https://github.com/kids-first/kf-somatic-workflow/pull/173

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

File: .github/scripts/workflow_to_docker_table.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+def parse_command():
+    """Function to parse the command line input
+    Args:
+        None
+    Return:
+        Namespace: returns the args as a standard Namespace object
+    """
+    parser = argparse.ArgumentParser(description='Generate Table of Tools and Their Dockers from a CWL Workflow')
+    parser.add_argument('input_file')
```
Comment (migbro) [inline .github/scripts/workflow_to_docker_table.py:14]: Does this syntax make it a named positional arg?

**Review by sickler-alex — APPROVED** [review]

File: .github/scripts/workflow_to_docker_table.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if not pathdict: pathdict = {}
+    if "workflows" not in pathdict: pathdict["workflows"] = []
+    if "tools" not in pathdict: pathdict["tools"] = []
+    inpath = pathlib.Path(filepath).resolve()
+    with open(inpath) as fh:
+        for line in fh:
+            if "run:" not in line: continue
+            relpath = inpath.parent / line.strip().split(' ')[-1]
+            respath = relpath.resolve()
+            if respath in pathdict["workflows"] or respath in pathdict["tools"]: continue
```
Comment (migbro) [inline .github/scripts/workflow_to_docker_table.py:39]: Might have to double check, but there might be older repos in which it's just `workflow` and not `workflows`

**Review by migbro — APPROVED** [review]
This is great! Just a couple minor Qs/comments

File: .github/scripts/workflow_to_docker_table.py (hunk 1 above)
Comment (dmiller15) [reply .github/scripts/workflow_to_docker_table.py:14]: Correct

File: .github/scripts/workflow_to_docker_table.py (hunk 2 above)
Comment (dmiller15) [reply .github/scripts/workflow_to_docker_table.py:39]: Yeah this doesn't really care. The only thing I truly check for is `*/tools/*` in line 40. This `pathdict` is just a custom object for this function. I could call it `pathdict["foo"]` and `pathdict["bar"]` and it wouldn't matter.

**Pull request merged**